app/code/community/Vindi/Subscription/Helper/Order.php

```diff
@@ -80,6 +80,15 @@ public function getOrder($data)
 		}
 
 		if (! $order || ! $order->getId()) {
+
+			# Ignora evento se for o primeiro ciclo de uma assinatura via cartão de crédito;
+			# Com exceção de transações com suspeita de fraude, 
+			# o Magento exclui o pedido caso o pagamento seja imediatamente rejeitado.
+			# Desse modo, não é possível realizar alterações no pedido
+			if (reset($data['bill']['charges'])['payment_method']['type'] == 'PaymentMethod::CreditCard' &&
```

Comment:
Qual evento vc se refere? Pensando aqui, se o Magento já não encontra o pedido, precisa ter alguma outra lógica nisso?

E não deveria retornar false?

Reply:
Nessa condição em específico não.
Contextualizando:
### Cenário Atual
- Tento realizar uma compra no Magento (com cartão inválido)
- A Vindi cancela automaticamente o pedido no Magento (fluxo de pagamento rejeitado)
- A Vindi dispara um _webhook_ (**charge_rejected**)
  - O pedido não existe e não é possível processar o webhook
  - É iniciado o fluxo de retentativas (envio de webhooks por até 48 horas)

### Cenário sujerido
- Tento realizar uma compra no Magento (com cartão inválido)
- A Vindi cancela automaticamente o pedido no Magento (fluxo de pagamento rejeitado)
- A Vindi dispara um _webhook_ (**charge_rejected**)
  - O pedido não existe e não é possível processar o webhook
  - O webhook é aprovado (se o método de pagamento for cartão de crédito **e** for o primeiro ciclo de uma assinatura)
## 
O método de pagamento retorna as informações da transação imediatamente, sendo assim, não é necessário o processamento do Webhook.